### PR TITLE
Refactor Autotools build

### PR DESCRIPTION
**Pull request by amitdo**
Use non-recursive make.

### CODE REVIEW

**Comment by amitdo** [comment]
This need a rebase.

File: configure.ac [hunk 1]
```diff
@@ -288,57 +283,40 @@ AC_ARG_ENABLE([debug],
...
 AC_MSG_RESULT([$debug])
 if test x"$debug" = x"yes"; then
-    AM_CXXFLAGS="$AM_CXXFLAGS -g -Wall -O0 -DDEBUG"
-    AM_CPPFLAGS="$AM_CPPFLAGS -g -Wall -O0 -DDEBUG"
+    AM_CXXFLAGS="$AM_CXXFLAGS -g -O0"
+    AM_CPPFLAGS="$AM_CPPFLAGS  -DDEBUG -Wall"
 else
-    AM_CXXFLAGS="$AM_CXXFLAGS -O2 -DNDEBUG"
-    AM_CPPFLAGS="$AM_CPPFLAGS -O2 -DNDEBUG"
+    AM_CXXFLAGS="$AM_CXXFLAGS -O1"
```
Comment (stweil) [inline configure.ac]: Who does the performance change with `-O1` instead of `-O2`?

Comment (amitdo) [inline configure.ac]: I did it for testing purpose. 

Forgot to change it back to -O2.

**Comment by amitdo** [comment]
I'm closing it for now.

**Pull request closed without merging**